Sources/_SwiftSyntaxTestSupport/Syntax+Assertions.swift

```diff
@@ -73,12 +75,14 @@ public func XCTAssertHasSubstructure(
 }
 
 /// See `SubtreeMatcher.assertSameStructure`.
-public func XCTAssertHasSubstructure(
-  _ actualTree: SyntaxProtocol,
-  _ expected: SyntaxProtocol,
+public func XCTAssertHasSubstructure<ActualTree, ExpectedTree>(
```

Comment:
Can’t this just be `<ActualTree: SyntaxProtocl, ExpectedTree: SyntaxProtocol>` instead of the `where` clause?

Reply:
It can, but that's 57 characters of generic signature on top of the 38 here which is just really cluttered.

Reply:
OK, I think I prefer declaring the requirement where the generic parameters are defined. But if you don't, that's fine.